docs/java-tutorial-2-graphx.md

```diff
@@ -0,0 +1,170 @@
+
+```bash
+python -m venv pyspark_venv
+source pyspark_venv/bin/activate
+pip3 install graphscope
+venv-pack -o pyspark_venv.tar.gz
+```
+
+Now, `pyspark_venv.tar.gz` contains neccessary enviroments graphscope need. Every time
+you submit a job to your spark cluster, remember to upload this pack.
+
+```bash
+export PYSPARK_DRIVER_PYTHON=python # Do not set in cluster modes.
+export PYSPARK_PYTHON=./environment/bin/python
+spark-submit --archives pyspark_venv.tar.gz#environment ...
+```
+
+## Run example GraphX apps
+
+Several GraphX algorithms are also contained in [grape-demo.jar](https://graphscope.oss-cn-beijing.aliyuncs.com/jar/grape-demo-0.18.0-shaded.jar).
+You can have a try to run these graphx algorithm on GraphScope.
+
+First, you also need to prepare the dataset.
+```bash
+git clone -b master --single-branch --depth=1 https://github.com/7br/gstest.git ${GS_TEST_DIR}
+```
+
+Different from Giraph-on-Graphscope, for GraphX-GraphScope integration, we need to submit jobs to spark cluster, not submit with GraphScope python client.
+
+### Prepare Environment
+
+The GraphX-GraphScope integration depends on serveral environment variables. We provide a shell script, user shall `source` this script before submit a graphx job.
+
+```bash
+
+if [[ "$GRAPHSCOPE_HOME"x == x ]];
+then
+    echo "No GRAPHSCOPE_HOME found, infer from python package"
+    site_path=`python3 -m site --user-site`
+    export GRAPHSCOPE_HOME=${site_path}/graphscope.runtime/
+    echo "Infered GRAPHSCOPE_HOME "${gs_runtime}
+fi
+
+if [[ "$SPARK_HOME"x == x ]];
+then
+    echo "[warn] No SPARK_HOME found!"
+fi
+
+GRAPHX_GRAPE_SDK=`ls ${GRAPHSCOPE_HOME}/lib/grape-graphx-*.jar`
+GRAPE_RUNTIME_JAR=`ls ${GRAPHSCOPE_HOME}/lib/grape-runtime-*.jar`
+# This env point to the directory where the output for llvm4jni run.sh on Grape-runtime.jar resides
+class_path=${GRAPE_RUNTIME_JAR}:${GRAPHX_GRAPE_SDK}
+
+#include jars in spark/jars
+if [ ! -z "${SPARK_HOME}" ]; then
+   jars=`ls ${SPARK_HOME}/jars`
+   for jar in ${jars}; do
+       class_path=${class_path}:${SPARK_HOME}/jars/${jar}
+   done
+fi
+
+jvm_version=$(${JAVA_HOME}/bin/javac -version 2>&1 | awk -F ' ' '{print $2}' | awk -F '.' '{print $1}')
+if  [[ "${jvm_version}" == *"11"* ]];then
+    _JVM_OPTS="-Xlog:gc*"
+    _JVM_OPTS=${_JVM_OPTS}" --illegal-access=warn"
+else 
+    _JVM_OPTS="-XX:+PrintGCDateStamps"
+    _JVM_OPTS=${_JVM_OPTS}" -XX:+PrintGCDetails"
+fi
+
+export GRAPE_JVM_OPTS="-Dcom.alibaba.fastffi.rvBuffer=2147483648 -XX:+StartAttachListener -XX:+PreserveFramePointer \
+-XX:+UseG1GC -XX:G1HeapRegionSize=32M \
+-XX:+PrintGCDetails -XX:+PrintGCDateStamps \
+-XX:+UnlockDiagnosticVMOptions -XX:LoopUnrollLimit=1 -XX:-TieredCompilation \
+-Djava.library.path=${GRAPHSCOPE_HOME}/lib \
+-Djava.class.path=${class_path}"
+```
+
+### Submit to Spark
+
+```bash
+export EXTRA_CP=${GRAPHSCOPE_HOME}/lib/:${GRAPE_RUNTIME_JAR}:${GRAPHX_SDK_JAR}
+
+./bin/spark-submit --verbose \
```

Comment:
is it possible  we generate this command for copy/paste

Reply:
Maybe I will export this as env variable in cli.